openedx/core/djangoapps/geoinfo/middleware.py

```diff
@@ -31,7 +29,7 @@ def process_request(self, request):
 
         Store country code in session.
         """
-        new_ip_address = get_real_ip(request)
+        new_ip_address, _ = get_client_ip(request, request_header_order=['X_FORWARDED_FOR', 'HTTP_X_FORWARDED_FOR'])
```

Comment:
From MDN Docs

> The X-Forwarded-For (XFF) header is a de-facto standard header for identifying the originating IP address of a client connecting to a web server through an HTTP proxy or a load balancer. When traffic is intercepted between clients and servers, server access logs contain the IP address of the proxy or load balancer only. To see the original IP address of the client, the X-Forwarded-For request header is used.

and this is precisely what we want here imo.

Reply:
If I'm reading the docs for `django-ipware` correctly, I think you don't need to change the order of the headers here.  It will go through all the headers [listed here](https://github.com/un33k/django-ipware#precedence-order) and I don't think we need to change the precedence.  I think the pattern of use here should be the same as in all the other places you've updated.

Reply:
There is some history behind this change. Initially, we were using it just like other places but that it caused an infinite loop of rendering 500 pages problem Jeremy mentioned [here](https://github.com/edx/edx-platform/pull/24827#discussion_r480324344), @awais786 figured out that this was due to `get_client_ip` returning local IP 127.0.0.0 from middleware in tests as opposed to returning None as the old method `get_real_ip` did. So we changed the header to get the original IP address as we always intended when using `get_real_ip`.

Reply:
@feanil I've reverted this change and used the same patterns as other places, and for the problem caused in tests for local IPs I've ignored them out in condition [here](https://github.com/edx/edx-platform/pull/24827/files#diff-c38dd29d1eb6cb48403c16cc5f45d2f08b704e4a7f2fcfab8222f92d313ab423R39), please have a look again.